### PR TITLE
lakectl: fixed bug in argument validation

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

**Review by eden-ohana — APPROVED** [review]

**Review by arielshaqed — APPROVED** [review]
Thanks!

AFAICT f7012ddf8edd31877af42448bdffc51bd950e71a introduced this regression.  Can we have a test that will catch the next time we do something like this?

Refs are subtle, and this is a really bad UX.

**Comment by nopcoder** [comment]
@johnnyaug is it possible to add the test value that will break the previous implementation?

**Comment by johnnyaug** [comment]
@nopcoder done

**Review by nopcoder — APPROVED** [review]
LGTM

File: esti/lakectl_test.go [hunk 1]
```diff
@@ -137,6 +137,23 @@ func TestLakectlCommit(t *testing.T) {
...
+		"REPO":    repoName,
+		"STORAGE": storage,
+		"BRANCH":  mainBranch,
+		"TAG":     validTagName,
+	}
+	invalidBranchName := "my.invalid.branch"
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" repo create lakefs://"+repoName+" "+storage, false, "lakectl_repo_create", vars)
+	RunCmdAndVerifyFailureWithFile(t, Lakectl()+" branch create lakefs://"+repoName+"/"+invalidBranchName+" --source lakefs://"+repoName+"/"+mainBranch, false, "lakectl_branch_create_invalid", vars)
+	RunCmdAndVerifySuccessWithFile(t, Lakectl()+" tag create lakefs://"+repoName+"/"+validTagName+" lakefs://"+repoName+"/"+mainBranch, false, "lakectl_tag_create", vars)
+}
```
Comment (arielshaqed) [inline esti/lakectl_test.go:155]: Neat!

Please also test `main~3`, `main~3^2^^`, and a few similar ones?

**Review by arielshaqed — APPROVED** [review]
THANKS!

**Pull request merged**